### PR TITLE
Add Neuralwatt provider

### PR DESCRIPTION
**Pull request by jrimmer**
## Summary

CodexBar can now query Neuralwatt's official `GET /v1/quota` endpoint with API-key authentication and render prepaid USD credit balance. The provider is wired through settings, config/environment resolution, managed token accounts, CLI, widgets, and provider documentation.

## Maintainer audit

Exact head: `b00c39edf`

Rebased onto current `main` at `129f6962e`; contributor authorship remains preserved in `1e0e869fc`.

Maintainer hardening on top of the contributor branch:

- preserves transport cancellation instead of converting it to a provider error;
- injects transport in tests and verifies endpoint overrides are rejected before secret lookup;
- rejects malformed successful quota responses and treats known zero-credit balances as exhausted;
- removes unrelated generated documentation churn and broken dashboard/status links;
- refreshes the generated parser hash;
- avoids a duplicate static credits section while keeping the prepaid balance in the primary metric;
- serializes app and CLI multi-account quota refreshes at Neuralwatt's documented one-request-per-second rate, preserving prior snapshots on cancellation;
- recognizes `NEURALWATT_API_KEY` in diagnostics and hides the unsupported generic Spend lane;
- documents that subscription kWh quota is separate from prepaid credit balance and is not currently shown.

Provider isolation audit: Neuralwatt identity, plan, and usage fields come only from the Neuralwatt response. API-key resolution is provider-scoped. No browser-cookie or keychain path is added.

## Official contract and product decision

Neuralwatt's [quota API documentation](https://portal.neuralwatt.com/docs/api/quota) confirms that prepaid credit balance and an active subscription's kWh allowance are separate quota pools. It also notes that lifetime USD cost can exceed prepaid credits because subscription usage is billed separately.

This branch deliberately keeps the contributor's balance-only product semantics. That is the smallest UI change, but it can show `$0` / exhausted while an active subscription still has usable kWh. Surfacing both pools is more accurate for subscription users, at the cost of adding a second unit and quota window to the menu model. Recommendation: **do not merge until we surface active subscription allowance alongside prepaid credits, or explicitly accept balance-only semantics as the product contract.**

## Runtime proof

The contributor supplied redacted live CLI output from the original branch, confirming the endpoint/auth path. No authorized Neuralwatt credential is available in the maintainer environment, so the rebased exact head could not be independently exercised live. The original proof therefore remains useful but is not exact-head proof.

## Validation

- exact-head focused Neuralwatt, multi-account rate-limit/cancellation, diagnostics, and provider-picker suites: passed;
- exact-head `make check`: passed;
- exact-head full `make test`: 44/44 shards passed;
- exact-head structured autoreview: no accepted/actionable findings;
- exact-head `./Scripts/package_app.sh`: passed and produced `CodexBar.app`;
- `git diff --check origin/main...HEAD`: passed;
- exact-head GitHub CI [run 28517756757](https://github.com/steipete/CodexBar/actions/runs/28517756757): Linux builds and lint passed; the four macOS shards and aggregate job were canceled by maintainer coordination to free runner capacity. No exact-head CI failure was observed, but this is not a terminal-green rollup.

## Integration state

The branch is mergeable against current `main`, including the landed cost-scanner changes from #1797 and the subsequent widget fix. Unrelated behind drift is intentionally preserved; rebase only if a real conflict or semantic overlap appears. Exact local proof is complete, while CI remains intentionally canceled pending the product/live-proof decision. No changelog entry is included on the contributor branch; the maintainer should add one with contributor thanks if the product decision is resolved and the PR is landed.

---

[Compound Engineering](https://github.com/EveryInc/compound-engineering-plugin)
Pi

### CODE REVIEW

File: Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattUsageFetcher.swift [hunk 1]
```diff
@@ -0,0 +1,430 @@
...
+            accountEmail: nil,
+            accountOrganization: nil,
+            loginMethod: self.displayLoginMethod)
+
+        return UsageSnapshot(
+            primary: primary,
+            secondary: nil,
+            tertiary: nil,
+            extraRateWindows: extras.isEmpty ? nil : extras,
+            subscriptionRenewsAt: nil,
```
[the remark this answers is not in the document]
Comment (jrimmer) [reply Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattUsageFetcher.swift:264]: Thanks for the review. I verified this against the product decision for Neuralwatt: it should be presented as a DeepSeek-style credit-exhaustion provider, not a subscription/reset provider. The live CLI proof also confirms the rendered shape is a primary credit balance with `secondary: null`, `tertiary: null`, and no renewal/reset window.

I am intentionally declining the suggested subscription-kWh window because surfacing `subscription.current_period_end` or kWh fields as capacity would reintroduce the confusing reset/renewal semantics the provider owner explicitly asked us to remove. The code still decodes subscription timestamps defensively, but the UI keeps Neuralwatt focused on the dollar balance and optional per-key allowance only.

File: Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattUsageFetcher.swift [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+        self.currentMonthEnergyKWh = currentMonthEnergyKWh
+        self.subscription = subscription
+        self.keyAllowance = keyAllowance
+        self.rateLimitTier = rateLimitTier
+        self.updatedAt = updatedAt
+    }
+
+    public var creditUsedPercent: Double {
+        guard let used = self.effectiveUsedCredits, let total = self.effectiveTotalCredits, total > 0 else {
+            return 0
```
[the remark this answers is not in the document]
Comment (jrimmer) [reply Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattUsageFetcher.swift:191]: Good catch — fixed in 31ebe870. Known zero-credit responses now return `creditUsedPercent == 100` when `credits_remaining_usd` is zero and there is no positive total, so an exhausted account cannot render as healthy.

I also added a regression test (`marks known zero credit balance as exhausted`) covering `credits_remaining_usd: 0` and `total_credits_usd: 0`. Validation: `swift test --filter NeuralWattUsageFetcherTests`, `swift test --filter MenuCardNeuralWattTests`, `swift build`, and `git diff --check upstream/main...HEAD` all pass.

**Comment by jrimmer** [comment]
@clawsweeper re-review

Added redacted runtime proof from a live Neuralwatt CLI fetch to the PR body, and addressed the inline review feedback. Current head: 31ebe870.

File: Sources/CodexBarCore/TokenAccountSupportCatalog+Data.swift [hunk 3]
```diff
@@ -129,6 +129,13 @@ extension TokenAccountSupportCatalog {
             injection: .environment(key: ElevenLabsSettingsReader.apiKeyEnvironmentKey),
             requiresManualCookieSource: false,
             cookieName: nil),
+        .neuralwatt: TokenAccountSupport(
```
[the remark this answers is not in the document]
Comment (steipete) [reply Sources/CodexBarCore/TokenAccountSupportCatalog+Data.swift:132]: Fixed in `b00c39edf`: Neuralwatt account support declares the documented [one-second minimum delay](https://github.com/steipete/CodexBar/blob/b00c39edf95b6c9c493931a3bef52fe93c5595c7/Sources/CodexBarCore/TokenAccountSupportCatalog%2BData.swift#L132-L139), and both app and CLI multi-account refresh paths serialize requests with that delay. Cancellation preserves prior snapshots for unstarted accounts; dedicated rate-limit/cancellation regression tests plus focused tests and `make check` pass.

File: Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattProviderDescriptor.swift [hunk 4]
```diff
@@ -0,0 +1,50 @@
...
+        ProviderDescriptor(
+            id: .neuralwatt,
+            metadata: ProviderMetadata(
+                id: .neuralwatt,
+                displayName: "Neuralwatt",
+                sessionLabel: "Credits",
+                weeklyLabel: "Spend",
+                opusLabel: nil,
+                supportsOpus: false,
+                supportsCredits: true,
```
[the remark this answers is not in the document]
Comment (steipete) [reply Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattProviderDescriptor.swift]: Fixed in `a2e33776f` and preserved at exact head `b00c39edf`: Neuralwatt now sets [`supportsCredits: false`](https://github.com/steipete/CodexBar/blob/b00c39edf95b6c9c493931a3bef52fe93c5595c7/Sources/CodexBarCore/Providers/NeuralWatt/NeuralWattProviderDescriptor.swift#L15-L17), so the generic duplicate Credits section is not rendered. Focused provider/menu tests and `make check` pass.

**Comment by steipete** [comment]
Carried forward and merged in #2220 as 82f161bd1ffe3fa0712537559a366aa2712a0be3.

The landed implementation preserves the original provider work and live-proof findings from @jrimmer, then adds @joeVenner's carry-forward integration and maintainer hardening. It now presents subscription kWh and prepaid balance as separate pools, handles the key-level blocked state, uses the shared transient retry policy, and includes focused synthetic/captured payload tests.

Thanks @jrimmer for the original implementation and live proof, and @joeVenner for carrying it forward.
